pageserver/src/tenant/timeline.rs

```diff
@@ -1859,19 +1859,17 @@ impl Timeline {
             "aborted because task_mgr shutdown requested".to_string()
         };
 
-        loop {
-            tokio::select! {
-                res = &mut calculation => { return res }
-                reason = timeline_state_cancellation => {
-                    debug!(reason = reason, "cancelling calculation");
-                    cancel.cancel();
-                    return calculation.await;
-                }
-                reason = taskmgr_shutdown_cancellation => {
-                    debug!(reason = reason, "cancelling calculation");
-                    cancel.cancel();
-                    return calculation.await;
-                }
+        tokio::select! {
+            res = &mut calculation => { return res }
+            reason = timeline_state_cancellation => {
+                debug!(reason = reason, "cancelling calculation");
+                cancel.cancel();
+                return calculation.await;
+            }
+            reason = taskmgr_shutdown_cancellation => {
+                debug!(reason = reason, "cancelling calculation");
+                cancel.cancel();
+                return calculation.await;
```

Comment:
Can we also remove these returns?

Reply:
we can! i suppose clippy's explicit return lint is turned off in macro expanded code, OR there is something in the macro expansion that makes it not fire.

pushed a commit to remove them.